`Communication`: Add last seen message separator to each channel

cremertim opened the pull request:
### Checklist
#### General
<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).



#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.


### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
n courses with active discussions, users frequently switch between multiple conversation threads. Without a clear visual indicator, it's hard to tell where they left off or which messages are new. This often leads to confusion, missed messages, or the need to scroll manually and guess the reading position.


### Description
<!-- Describe your changes in detail -->
This change introduces a visual separator for the last seen message in each conversation channel.
The separator allows users to easily identify which messages are new since their last visit.


### Steps for Testing
<!-- Please describe in detail how reviewers can test your changes. Make sure to take all related features and views into account! Below is an example that you can refine. -->
Prerequisites:
- 2 Users in the same course
- 1 Course with communication enabled
-  2 Browser sessions with different users (e.g use a Incognito Tab)

1. Log in to **Artemis** as *User A*.
2. Navigate to the **Communication** section of a course.
3. Post a new message in an existing conversation.
4. Log in as **User B** (who has not yet read the new message).
5. Navigate to the same conversation.
6. Verify that a **"new message" separator** appears above the unread message.
7. Scroll slightly upwards so that the unread message is only **partially visible**.
8. Confirm that the **"Jump to first unread message"** button appears.
9. Click the button.
10. Verify that the view scrolls to make the **first unread message fully visible** at the bottom of the container.

### Testserver States
You can manage test servers using [Helios](https://helios.aet.cit.tum.de/). Check environment statuses in the [environment list](https://helios.aet.cit.tum.de/repo/69562331/environment/list). To deploy to a test server, go to the [CI/CD](https://helios.aet.cit.tum.de/repo/69562331/ci-cd) page, find your PR or branch, and trigger the deployment.

### Review Progress
<!-- Each PR should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files modified in this PR here. You can use `supporting_script/code-coverage/generate_code_cov_table/generate_code_cov_table.py` to automatically generate the coverage table from the corresponding artefacts of your branch (follow the ReadMe for setup details). -->
<!-- Alternatively you can execute the tests locally (see build.gradle and package.json) or look into the corresponding artefacts. -->
<!-- The line coverage must be above 90% for changes files, and you must use extensive and useful assertions for server tests and expect statements for client tests. -->
<!-- Note: Confirm in the last column that you have implemented extensive assertions for server tests and expect statements for client tests. -->

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| conversation-messages.component.ts | 94.55% | ✅              |

## Summary by CodeRabbit

* **New Features**
  * Added a visual "New" separator before the first unread message in conversation threads.
  * Introduced a sticky "New messages" button to quickly jump to the first unread message when not in view.
  * Enhanced tracking and display of unread messages for improved navigation and clarity.

* **Style**
  * Updated styles to support the new separator and sticky button for unread messages.

* **Bug Fixes**
  * Improved handling of message visibility and scrolling behavior for unread messages.

* **Tests**
  * Added unit tests covering unread message detection, visibility logic, and scrolling actions.

* **Documentation**
  * Added new English and German translations for the "New" separator and "Jump to first unread message" button.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added a sticky "New messages" button above the message list for quick navigation to unread messages.
  * Introduced a visual "New" separator before the first unread message to clearly mark unread content.
  * Enhanced message list styling for improved readability and separation of new messages.
  * Added support for new English and German translations related to unread messages.

* **Bug Fixes**
  * Improved detection and scrolling behavior to ensure users can easily jump to the first unread message.

* **Tests**
  * Expanded unit tests to cover unread message detection, visibility checks, and scroll behavior.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jonas-de reviewed (changes requested):
Tested on TS2 using Safari:
The separator and jump to first unread message button only appears when the channel is already open when the message is sent.
Also, the button does not disappear once one reaches the message by manual scrolling or clicks the button, which is a bit confusing. I think, once the message is fully visible the system should mark it as read and remove the separator and the button.
Ideally, I would expect the chat to automatically scroll down to the newest message in case I'm already at the bottom of the conversation. However, this would probably be in the scope of a follow up PR

cremertim commented:
Hi @jonas-de , 
I fixed the issue when changing the channel. It should now properly display previously unread messages.

I was able to reproduce the bug with the button. I think it was related to a nearly empty channel, so i did not properly trigger the observation of visible post. I fixed it in my last commit. Can you please retest it?

badkeyy reviewed (changes requested):
Tested on TS2 nearly works as expected, however when I press "Jump to first unread message") it jumps to the 2nd newest message (See screen recording below).

https://github.com/user-attachments/assets/072ffb57-34b9-4d03-8c8a-42ee827a5abd

cremertim commented:
@badkeyy thanks for testing. This was related to a potential margin at the bottom of the first unseen post if and only if there is only a single new post. I adjusted the offset so we only see the first visible post 👍

Predixx reviewed (dismissed):
Tested on ts2. Worked for me and looks good!

konrad2002 reviewed (changes requested):
<img width="835" height="592" alt="grafik" src="https://github.com/user-attachments/assets/81c0b584-1f66-4301-99d8-5ca498b453dd" />
when using Safari, the jump button only brings me to the blue horizontal line and the message itself is not visible without manual scrolling (tested on TS3 in Safari)

anian03 reviewed (commented):
Tested on TS3, I noticed the following behavior on Safari: The message is not always fully visible, for me it only scrolls the message header (profile pic/name) into view and the message itself is cut off. For multiple unread messages I could not reproduce this behavior.

I noticed though that we still show this button after the user has sent messages into a channel, we probably should update the `lastReadDate` whenever the user sends a message (otherwise it is considered unread).

Is it intentional that the button is anchored to the top?

jonas-de reviewed (changes requested):
Tested on TS4 with Safari. The button is now hidden when one scrolls below the separator which is nice. Also, the scrolling has been optimized, but I recognized an issue with long messages. The button scrolls to the bottom of the message so the message header or start might not be visible. Also, I would prefer if the separator or at least the button permanently disappears after one scrolls to the new message because then it is not unread anymore

SindiBuklaji reviewed (approved):
Works as described on TS2 👍

anian03 reviewed (approved):
Tested on TS2, described features work. I still get the jumping behavior, but it is not unusable so we could look into it in a follow up maybe.

Maybe it is related to the button being part of the scrollable container? It being created/destroyed technically changes how many elements are above in the scrollable area. You could try to see if moving the button outside of that list solves the issue

eylulnc reviewed (approved):
Tested on TS2, works as expected
